gotham/src/helpers/http/response/mod.rs

```diff
@@ -225,21 +247,14 @@ pub fn extend_response(state: &State, res: &mut Response, status: StatusCode, bo
         );
     }
 
-    match body {
-        Some((body, mime)) => {
-            set_headers(state, res, Some(mime), Some(body.len() as u64));
-            res.set_status(status);
-
-            match *Method::borrow_from(state) {
-                Method::Head => (),
-                _ => res.set_body(body),
-            };
-        }
-        None => {
-            set_headers(state, res, None, None);
-            res.set_status(status);
-        }
-    };
+    match mime {
```

Comment:
This would look a little better like this (IMO, anyway):

```
if let Some(mime) = mime {
  *builder = builder.header(CONTENT_TYPE, mime.as_ref());
}
```